### PR TITLE
Mark __init_subclass__ as classmethod [2.0 branch]

### PR DESCRIPTION
**Pull request by rogalski**
Just a POC, needs test coverage

### CODE REVIEW

**Comment by degustaf** [comment]
Sorry, missed the POC comment when I reviewed this.

**Comment by rogalski** [comment]
@degustaf hopefully tomorrow

**Comment by PCManticore** [comment]
This looks good. Needs tests, of course. Also, is this targetted to 2.0? I am on the phone and can't really see the destination, but the changed file is ``tree/scoped_nodes.py``.

**Comment by rogalski** [comment]
@PCManticore yes, it targets 2.0 branch. Should it?

**Comment by PCManticore** [comment]
Probably no, it could target the master branch, which will be released much sooner than 2.0 is expected to be.

File: astroid/tests/unittest_builder.py [hunk 1]
```diff
@@ -718,6 +718,13 @@ def test_method_base_props(self):
         self.assertEqual(method.args.args, [])
         self.assertEqual(method.type, 'staticmethod')
 
+    @test_utils.require_version(minver='3.6')
```
Comment (PCManticore) [inline astroid/tests/unittest_builder.py]: We need to drop this test.

**Comment by PCManticore** [comment]
Looks good. Feel free to merge once you remove the extraneous test.

**Pull request merged**